### PR TITLE
Feature/expose signed uri

### PR DESCRIPTION
**Pull request by chinyong**

### CODE REVIEW

File: lib/viki/core/base.rb [hunk 1]
```diff
@@ -84,6 +84,10 @@ def fetch(url_options = {}, &block)
         fetcher
       end
 
+      def signed_http_get_uri(url_options = {}, body = nil)
```
Comment (saarangsoltani) [inline lib/viki/core/base.rb]: 'get_signed_http_uri' looks like a better name for this method

Comment (Weiyuan-Lane) [inline lib/viki/core/base.rb]: I agree. 'http' too seems to hold little meaning in the context of the method. changing to 'get_signed_uri' seems to be ideal

Comment (ntd251) [inline lib/viki/core/base.rb]: According to what I understand from Chin Yong, `http_get` is a phrase referring to HTTP GET method.

File: spec/viki/core/base_spec.rb [hunk 2]
```diff
@@ -135,6 +135,19 @@
     its(:query_values) { should have_key("sig") }
   end
 
+  describe "#get_signed_uri" do
+    let(:uri) { Addressable::URI.parse "http://example.com" }
+    let(:body) { "" }
+
+    it 'makes sure input params\' values are preserved' do
```
Comment (allenlsy) [inline spec/viki/core/base_spec.rb]: in this case, you can use double quotes

File: spec/viki/core/base_spec.rb [hunk 3]
```diff
@@ -135,6 +135,19 @@
     its(:query_values) { should have_key("sig") }
   end
 
+  describe "#get_signed_uri" do
+    let(:uri) { Addressable::URI.parse "http://example.com" }
+    let(:body) { "" }
```
Comment (allenlsy) [inline spec/viki/core/base_spec.rb]: `body` is not used. Either remove it for now or use it ?

**Pull request merged**